crowdsec-docs/docs/bouncers/nginx.mdx

```diff
@@ -79,102 +94,314 @@ If you are on a mono-machine setup, the `crowdsec-nginx-bouncer` install script
+#those apply for "captcha" action
+# reCAPTCHA Secret Key
+SECRET_KEY=
+# reCAPTCHA Site key
+SITE_KEY=
+CAPTCHA_TEMPLATE_PATH=/var/lib/crowdsec/lua/templates/captcha.html
+CAPTCHA_EXPIRATION=3600
+```
+
+
+### NGINX Configuration
+
+
+The bouncer NGINX configuration is located in `/etc/nginx/conf.d/crowdsec_nginx.conf` :
+
+```bash title="/etc/nginx/conf.d/crowdsec_nginx.conf"
+lua_package_path '/usr/lib/crowdsec/lua/?.lua;;';
+lua_shared_dict crowdsec_cache 50m;
+resolver 8.8.8.8 ipv6=off;
+lua_ssl_trusted_certificate /etc/ssl/certs/ca-certificates.crt;
+init_by_lua_block {
+        cs = require "crowdsec"
+        local ok, err = cs.init("/etc/crowdsec/bouncers/crowdsec-nginx-bouncer.conf", "crowdsec-nginx-bouncer/v0.0.7")
+        if ok == nil then
+                ngx.log(ngx.ERR, "[Crowdsec] " .. err)
+                error()
+        end
+        ngx.log(ngx.ALERT, "[Crowdsec] Initialisation done")
+}
+
+access_by_lua_block {
+        local cs = require "crowdsec"
+        cs.Allow(ngx.var.remote_addr)        
+}
+```
+
+
+The bouncer uses [lua_shared_dict](https://github.com/openresty/lua-nginx-module#lua_shared_dict) to share cache between all workers.
+
+If you want to increase the cache size you need to change this value `lua_shared_dict crowdsec_cache 50m;`.
+
+:warning: Do not rename the `crowdsec_cache` shared dict, else the bouncer will not work anymore.
 
-#### Requirements
+To make HTTP request in the bouncer, we need to set a `resolver` in the configuration. We choose `8.8.8.8` since we query `google.com` for the captcha verification, but you can replace it with a valid one.
 
-The following packages are required :
+To make secure HTTP request in the bouncer, we need to specify a trusted certificate (`lua_ssl_trusted_certificate`). You can also change this with a valid one.
 
-- lua
-- lua-sec
-- libnginx-mod-http-lua
 
-##### Debian/Ubuntu
+### Setup captcha
+> Currently, only reCAPTCHA v2 is supported.
+
+If you want to use reCAPTCHA with your Nginx Bouncer, you must provide a reCAPTCHA Site key and Secret key in your bouncer configuration.
+
+Edit `etc/crowdsec/bouncers/crowdsec-nginx-bouncer.conf` and configure the following options:
 
 ```bash
-sudo apt-get install lua5.3 libnginx-mod-http-lua lua-sec
+SECRET_KEY=
+SITE_KEY=
+CAPTCHA_TEMPLATE_PATH=/var/lib/crowdsec/lua/templates/captcha.html
+CAPTCHA_EXPIRATION=3600
 ```
 
-Download the following 2 repositories:
+Restart Nginx.
+
+You can add a decisions with a type `captcha` to check if it works correctly:
 
-- [`lua-cs-bouncer`](https://github.com/crowdsecurity/lua-cs-bouncer):
 ```bash
-git clone https://github.com/crowdsecurity/lua-cs-bouncer.git
+sudo cscli decisions add -i <IP_TO_TEST> -t captcha
+```
+
+## FAQ
+
+### Why remediation are not apply instantaneous
+
+In stream mode, the bouncer will lunch the internal timer to pull the local API after the first request. After installing the bouncer, just do a simple curl on your website to trigger the puller timer.
```

Comment:
In stream mode, the bouncer will launch an internal timer to pull the local API at the first request. So the cache won't be refreshed until the first request hits the service.